examples/04_New Models.ipynb

```diff
@@ -835,6 +835,190 @@
+    "class ThrownObject_MM(PrognosticsModel):\n",
+    "    # Define the model properties, this is exactly the same as for a regular model.\n",
+    "\n",
+    "    inputs = []  # no inputs, no way to control\n",
+    "    states = [\n",
+    "        'x',    # Position (m) \n",
+    "        'v'    # Velocity (m/s)\n",
+    "        ]\n",
+    "    outputs = [\n",
+    "        'x'     # Position (m)\n",
+    "    ]\n",
+    "    events = [\n",
+    "        'falling',  # Event- object is falling\n",
+    "        'impact'    # Event- object has impacted ground\n",
+    "    ]\n",
+    "\n",
+    "    is_vectorized = True\n",
+    "\n",
+    "    # The Default parameters. Overwritten by passing parameters dictionary into constructor\n",
+    "    default_parameters = {\n",
+    "        'thrower_height': 1.83,  # m\n",
+    "        'throwing_speed': 40,  # m/s\n",
+    "        'g': -9.81,  # Acceleration due to gravity in m/s^2\n",
+    "        'process_noise': 0.0  # amount of noise in each step\n",
+    "    }\n",
+    "\n",
+    "    # Add initialization \n",
+    "    def initialize(self, u = None, z = None):\n",
+    "        # Note: states are returned using StateContainer\n",
+    "        return self.StateContainer({\n",
+    "            'x': self.parameters['thrower_height'], \n",
+    "            'v': self.parameters['throwing_speed']})    \n",
+    "\n",
+    "    def output(self, x):\n",
+    "        # Note- states can still be accessed a dictionary\n",
+    "        return self.OutputContainer({'x': x['x']})\n",
+    "\n",
+    "    # This is actually optional. Leaving thresholds_met empty will use the event state to define thresholds.\n",
+    "    #  Threshold = Event State == 0. However, this implementation is more efficient, so we included it\n",
+    "    def threshold_met(self, x):\n",
+    "        return {\n",
+    "            'falling': x['v'] < 0,\n",
+    "            'impact': x['x'] <= 0\n",
+    "        }\n",
+    "\n",
+    "    def event_state(self, x): \n",
+    "        x_max = x['x'] + np.square(x['v'])/(-self.parameters['g']*2) # Use speed and position to estimate maximum height\n",
+    "        x_max = np.where(x['v'] > 0, x['x'], x_max) # 1 until falling begins\n",
+    "        return {\n",
+    "            'falling': np.maximum(x['v']/self.parameters['throwing_speed'],0),  # Throwing speed is max speed\n",
+    "            'impact': np.maximum(x['x']/x_max,0)  # then it's fraction of height\n",
+    "        }"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we need to define how the state transitions, and we'll use matrices to do so. Since we are working with a discrete version of the system now, we'll define the `next_state` method. \n",
+    "\n",
+    "In the following, we will use the matrix version for each variable, accessed with `.matrix`. Both `x.matrix` and `u.matrix` are column vectors, and `u.matrix` is in the same order as model.inputs."
```

Comment:
Let's mention somewhere how we're using it in next_state here, but this feature can also be used in other functions.